Sources/NIOExtras/FixedLengthDecoder.swift

```diff
@@ -0,0 +1,67 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+/// Error indicating that after an operation some unused bytes are left.
+public struct LeftOverBytesError: Error {
```

Comment:
@Lukasa I didn't know where to put this Error. Should I create a separate file?

Reply:
Yeah, a separate file. In fact, we have a weird pattern developing for this, which you can see in our [HTTP/2 repository](https://github.com/apple/swift-nio-http2/blob/master/Sources/NIOHTTP2/HTTP2Error.swift). Essentially, you create a top-level enum with no cases, and make this struct a member of that. You then also create a tag protocol and conform this struct to it.

Reply:
That said, for this you may do better to just make this error a nested type in `FixedLengthFrameDecoder`.